scheds/rust/scx_layered/src/main.rs

```diff
@@ -1966,54 +1970,78 @@ impl<'a> Scheduler<'a> {
+            let nr_available_cpus = available_cpus.count_ones();
+
+            // Open layers need the intersection of allowed cpus and
+            // available cpus.
+            layer.cpus.copy_from_bitslice(&available_cpus);
+            layer.nr_cpus = nr_available_cpus;
+            Self::update_bpf_layer_cpumask(layer, bpf_layer);
+        }
 
-            for (lidx, layer) in self.layers.iter().enumerate() {
-                self.nr_layer_cpus_ranges[lidx] = (
-                    self.nr_layer_cpus_ranges[lidx].0.min(layer.nr_cpus),
-                    self.nr_layer_cpus_ranges[lidx].1.max(layer.nr_cpus),
-                );
-            }
+        self.skel.maps.bss_data.fallback_cpu = self.cpu_pool.fallback_cpu as u32;
 
-            // Trigger updates on the BPF side.
-            let input = ProgramInput {
-                ..Default::default()
-            };
-            let prog = &mut self.skel.progs.refresh_layer_cpumasks;
-            let _ = prog.test_run(input);
+        for (lidx, layer) in self.layers.iter().enumerate() {
+            self.nr_layer_cpus_ranges[lidx] = (
+                self.nr_layer_cpus_ranges[lidx].0.min(layer.nr_cpus),
+                self.nr_layer_cpus_ranges[lidx].1.max(layer.nr_cpus),
+            );
+        }
 
-            // Update empty_layers.
-            let empty_layer_ids: Vec<u32> = self
-                .layers
-                .iter()
-                .enumerate()
-                .filter(|(_idx, layer)| layer.nr_cpus == 0)
-                .map(|(idx, _layer)| idx as u32)
-                .collect();
-            for i in 0..self.layers.len() {
-                self.skel.maps.bss_data.empty_layer_ids[i] =
-                    empty_layer_ids.get(i).cloned().unwrap_or(MAX_LAYERS as u32);
-            }
-            self.skel.maps.bss_data.nr_empty_layer_ids = empty_layer_ids.len() as u32;
+        // Trigger updates on the BPF side.
+        self.refresh_cpumasks_bpf();
+
+        // Update empty_layers.
+        let empty_layer_ids: Vec<u32> = self
+            .layers
+            .iter()
+            .enumerate()
+            .filter(|(_idx, layer)| layer.nr_cpus == 0)
+            .map(|(idx, _layer)| idx as u32)
+            .collect();
+        for i in 0..self.layers.len() {
+            self.skel.maps.bss_data.empty_layer_ids[i] =
+                empty_layer_ids.get(i).cloned().unwrap_or(MAX_LAYERS as u32);
+        }
+        self.skel.maps.bss_data.nr_empty_layer_ids = empty_layer_ids.len() as u32;
+
+        Ok(())
+    }
+
+    fn refresh_cpumasks(&mut self) -> Result<()> {
+        let targets = self.calc_target_nr_cpus();
+        let targets = self.weighted_target_nr_cpus(&targets);
+
+        let mut ascending: Vec<(usize, usize)> = targets.iter().copied().enumerate().collect();
+        ascending.sort_by(|a, b| a.1.cmp(&b.1));
+
+        let shrunk_layers = self.shrink_layers(&ascending, &targets)?;
```

Comment:
shrank?

Reply:
Apparently "shrank" is the past simple in UK English (https://dictionary.cambridge.org/dictionary/english/shrank) while "shrunk" is the past participle (https://dictionary.cambridge.org/dictionary/english/shrunk). Fortunately shrunk is both in US English so I don't have to learn what the difference is :)

Reply:
Oh, so the next line is:
```
        let grew_layers = self.grow_non_open_layers(&ascending)?;
```
It's a bit odd for one to be pp and the other p.